classes/form/CustomerAddressForm.php

```diff
@@ -154,10 +140,16 @@ public function submit()
 
         $this->setAddress($address);
 
-        return $this->getPersister()->save(
-            $address,
-            $this->getValue('token')
-        );
+        try {
+            return $this->getPersister()->save(
+                $address,
+                $this->getValue('token')
+            );
+        } catch (PrestaShopException $e) {
+            $this->errors[''][] = $this->translator->trans('Could not update your information, please check your data.', [], 'Shop.Notifications.Error');
```

Comment:
@Julievrz 
I found this generic message to say that there is something wrong in the form and we can't save it
Maybe you can suggest another one :) Thanks

Reply:
I tried, but I couldn't find better wording. I think this one is ok as it has a pretty logical domain.